source/tutorials/nix-language.md

```diff
@@ -0,0 +1,1602 @@
+
+### What do you need?
+
+- Familiarity with software development
+- Familiarity with Unix shell, to read command line examples <!-- TODO: link to yet-to-be instructions on "how to read command line examples" -->
+- [](./install-nix), to run the examples
+
+### How long does it take?
+
+- **No experience with functional programming: 2 hours**
+
+  Carefully work through all definitions and examples.
+
+  Make sure you understand them by reading all detailed explanations.
+
+- **Familiar with functional programming: 1 hour**
+
+  Learn syntax and semantics by examining definitions and examples, and read detailed explanations if in doubt.
+
+- **Proficient with functional programming: 30 minutes**
+
+  The Nix language is largely unsurprizing to experienced functional programmers.
+
+  Read through the guide to get accustomed to the syntax and language features.
+
+We recommend running all examples and playing with them to validate your assumptions and test what you have learned.
+
+### How to run the examples?
+
+- Every valid piece of Nix language code is a *Nix expression*.
+- Nix expressions can contain other Nix expressions, that is, they can be nested.
+- Evaluating a Nix expression produces a single value.
+- Every *Nix file* (`.nix`) contains a single Nix expression.
+
+:::{note}
+To *evaluate* means to transform an expression according to the language rules until no further simplification is possible.
+:::
+
+All examples in this guide are valid Nix expressions.
+Each one is followed by the expected evaluation result.
+
+The following example is a Nix expression adding two numbers:
+
+```nix
+1 + 2
+```
+
+    3
+
+#### Interactive evaluation
+
+Use [`nix repl`][nix-repl] to evaluate Nix expressions interactively (by typing them on the command line):
+
+```console
+nix repl
+```
+
+    Welcome to Nix 2.5.1. Type :? for help.
+
+```console
+nix-repl> 1 + 2
+```
+
+    3
+
+[nix-repl]: https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-repl.html
+
+#### Evaluating Nix files
+
+Use [`nix-instantiate --eval`][nix-instantiate] to evaluate the expression in a Nix file.
+
+```console
+echo 1 + 2 > file.nix
+
+nix-instantiate --eval file.nix
+```
+
+    3
+
+<details><summary>Detailed explanation</summary>
+
+The first command writes `1 + 2` to a file `file.nix` in the current directory.
+The contents of `file.nix` are now `1 + 2`, which you can check with
+
+```console
+cat file.nix
+```
+
+    1 + 2
+
+The second command runs `nix-instantiate` with the `--eval` option on `file.nix`, which reads the file and evaluates the contained Nix expression.
+The resulting value is printed as output.
+
+`--eval` is required to evaluate the file and do nothing else.
+If `--eval` is omitted, `nix-instantiate` expects the expression in the given file to evaluate to a special data type called *Derivation*, which we will come back to at the end of this guide in [](build-tasks).
```

Comment:
I'd argue it is special, because it's the whole point of Nix. It deserves special attention. Even if we should decide not to call it special here, I would like to emphasize that it is a value of a certain data type as opposed to the other three things a {store d,d,`d`}erivation could be.

Reply:
Fair enough, I mainly think "data type" is not right. How about

```suggestion
If `--eval` is omitted, `nix-instantiate` expects the expression in the given file to evaluate to a special value called *Derivation*, which we will come back to at the end of this guide in [](build-tasks).
```